src/Turtle/Options.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+-- TODO: documentation
+
+module Turtle.Options
+    ( Parser
+    , ParameterName(..)
+    , LongName(..)
+    , HelpMessage(..)
+    , options
+    , switch
+    , parameter
+    , parameterAuto
+    , parameterText
+    , parameterInteger
+    , parameterInt
+    , parameterDouble
```

Comment:
Yeah, `ArgumentName` would be more consistent

I also like the idea of using `argRead`